auth/signup.go

```diff
@@ -21,13 +22,13 @@ type User struct {
 
 const UserNameSpace = "users"
 
-func (a *auth) ValidateUser(u User) error {
+func (u *User) Validate(store cache.Store) error {
```

Comment:
@guacamole can we call this something like `checkForDuplicatesAndValidate()`?

Reply:
Hey @jay-dee7 , I think ValidateUser is good as for keeping it short. we can maybe extract the checking duplicate part and merge it in a future PR? LMK

Reply:
sure that works too